### ISSUE TITLE
Breakpoints opens read-only file

### ISSUE THREAD

**Issue by VandalPaulius** [microsoft/vscode-js-debug#480]
- VS Code Version:
- Log file (set `"trace": true` in launch config):

Steps to reproduce:

1. Have generic launch.json config like
```
{
      "type": "pwa-chrome",
      "request": "launch",
      "name": "Launch Chrome debug",
      "url": "http://localhost:8080",
      "webRoot": "${workspaceFolder}",
      "disableNetworkCache": true,
      "runtimeArgs": ["--incognito"],
}
```
2. Have VSCode Chrome Debugger extension enabled
3. Set breakpoint (or add `debugger`) anywhere in your code
4. Use website to hit the breakpoint
5. VSCode opens new read-only file that has similar path to the original, has all same breakpoints, but is read-only. Although one thing to note is read-only file path starts with `~/Desktop` while for original VSCode shows `/home/myUser/Desktop`

System is:
Ubuntu 19.10
Node v12.8.1
VSCode v1.45.1
Debugger for Chrome v4.12.7

Files are converted and served with Webpack and `webpack-plugin-serve`

Expected behaviour:
Debugger continues to use same file with full edit support

Edit:
https://github.com/VandalPaulius/code-examples I've created example repository for this

**Comment by connor4312** [comment]
Can you please upload the log file that `"trace": true` creates? Thanks!

**Comment by VandalPaulius** [comment]
https://github.com/VandalPaulius/code-examples I've created example repository for this, so would be easier to reproduce since `trace` option produced 13k line file :sweat_smile:

**Comment by connor4312** [comment]
Hi, I was about to check on this, but based on your last commit it looks like you got it working.

The key was setting the `webRoot`, which we use to resolve the relative paths that webpack produces by default (e.g. `webpack://./src/app.tsx`). By default this is the workspace folder, but if you have your application served from a subdirectory, you need to adjust for that in the webroot.

Something I will do in the near future is write some documentation for fixing breakpoint-related issues -- it's usually some variant of sourcemap/path misconfiguration.

**Comment by VandalPaulius** [comment]
Thanks for checking it :100: 
Yes, I got it working just now. Was about write it in here. As always it was user error :smile:
As you said it was sourcemap and webroot config issue. I will keep this repo on, so anyone that comes to this problem has some reference (even though setup in that repo is a bit unconventional)

**Comment by Slos** [comment]
Why has this issue been closed? This is still happening in version 1.47.3 (on Mac OS)

**Comment by connor4312** [comment]
Please open a new issue and fill in the template @Slos https://github.com/microsoft/vscode-js-debug/issues/new/choose
